dotnet/src/Agents/AzureAI/Extensions/AgentDefinitionExtensions.cs

```diff
@@ -18,15 +18,15 @@ internal static class AgentDefinitionExtensions
     /// <exception cref="InvalidOperationException"></exception>
     public static IEnumerable<ToolDefinition> GetAzureToolDefinitions(this AgentDefinition agentDefinition)
     {
-        return agentDefinition.Tools.Select<AgentToolDefinition, ToolDefinition>(tool =>
+        return agentDefinition.Tools?.Select<AgentToolDefinition, ToolDefinition>(tool =>
         {
             return tool.Type switch
             {
                 "code_interpreter" => new CodeInterpreterToolDefinition(),
                 "file_search" => new FileSearchToolDefinition(),
                 _ => throw new NotSupportedException($"Unable to create Azure AI tool definition because of unsupported tool type: {tool.Type}"),
```

Comment:
Should failing one tool, stop all the list of being created? 

Suggest getting those types and throw the list of types that are not recognized (or just mention the two that are allowed) in a single exception, if we have two or more, the caller will experience subsequent surprises.

```suggestion
                _ => throw new NotSupportedException($"Unable to enumerate Azure AI tool definitions unsupported tool type where found: {unsupportedTypes}"),
```

Reply:
Good suggestion, there are more tool types to be added in a later PR so I'll make that change when I have the full list